modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java

```diff
@@ -0,0 +1,549 @@
+import org.apache.ignite.internal.processors.query.IgniteSQLException;
+import org.apache.ignite.internal.processors.query.h2.SchemaManager;
+import org.apache.ignite.internal.processors.query.h2.opt.GridH2Table;
+import org.apache.ignite.internal.processors.query.stat.config.StatisticsColumnConfiguration;
+import org.apache.ignite.internal.processors.query.stat.config.StatisticsObjectConfiguration;
+import org.apache.ignite.internal.processors.subscription.GridInternalSubscriptionProcessor;
+import org.apache.ignite.internal.util.GridSpinBusyLock;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.X;
+import org.apache.ignite.thread.IgniteThreadPoolExecutor;
+import org.gridgain.internal.h2.table.Column;
+
+/**
+ *
+ */
+public class IgniteStatisticsConfigurationManager {
+    /** */
+    private static final String STAT_OBJ_PREFIX = "sql.statobj.";
+
+    /** */
+    private final IgniteStatisticsRepository repo;
+
+    /** Schema manager. */
+    private final SchemaManager schemaMgr;
+
+    /** */
+    private final StatisticsGatherer gatherer;
+
+    /** */
+    private final IgniteThreadPoolExecutor mgmtPool;
+
+    /** Logger. */
+    private final IgniteLogger log;
+
+    /** Distributed metastore. */
+    private volatile DistributedMetaStorage distrMetaStorage;
+
+    /** */
+    private volatile boolean started;
+
+    /** Busy lock on node stop. */
+    private final GridSpinBusyLock stopLock;
+
+    /** Monitor to synchronize changes repository: aggregate after collects and drop statistics. */
+    private final Object mux = new Object();
+
+    /** */
+    public IgniteStatisticsConfigurationManager(
+        SchemaManager schemaMgr,
+        GridInternalSubscriptionProcessor subscriptionProcessor,
+        GridCachePartitionExchangeManager exchange,
+        IgniteStatisticsRepository repo,
+        StatisticsGatherer gatherer,
+        IgniteThreadPoolExecutor mgmtPool,
+        GridSpinBusyLock stopLock,
+        Function<Class<?>, IgniteLogger> logSupplier
+    ) {
+        this.schemaMgr = schemaMgr;
+        log = logSupplier.apply(IgniteStatisticsConfigurationManager.class);
+        this.repo = repo;
+        this.mgmtPool = mgmtPool;
+        this.gatherer = gatherer;
+        this.stopLock = stopLock;
+
+        subscriptionProcessor.registerDistributedMetastorageListener(new DistributedMetastorageLifecycleListener() {
+            @Override public void onReadyForRead(ReadableDistributedMetaStorage metastorage) {
+                distrMetaStorage = (DistributedMetaStorage)metastorage;
+
+                distrMetaStorage.listen(
+                    (metaKey) -> metaKey.startsWith(STAT_OBJ_PREFIX),
+                    (k, oldV, newV) -> {
+                        // Skip invoke on start node (see 'ReadableDistributedMetaStorage#listen' the second case)
```

Comment:
All identifiers in the Javadoc comments must be placed within <tt></tt> or linked to, if possible, using {@link ClassName} or {@link ClassName#method} constructs.